contracts/allocators/LUSDAllocatorV2.sol

```diff
@@ -0,0 +1,277 @@
+        }
+
+        // 3.  If we have eth, convert to weth, then swap a percentage of it to LUSD.
+        uint256 ethBalance = address(this).balance; // Use total balance in case we have leftover from a prior failed attempt
+        bool swappedLUSDSuccessfully;
+        if (ethBalance > 0) {
+            // Wrap ETH to WETH
+            IWETH(wethAddress).deposit{value: ethBalance}();
+
+            uint256 wethBalance = IWETH(wethAddress).balanceOf(address(this)); //Base off of WETH balance in case we have leftover from a prior failed attempt
+            if (ethToLUSDRatio > 0) {
+                uint256 amountWethToSwap = (wethBalance * ethToLUSDRatio) / FEE_PRECISION;
+                uint256 amountLUSDMin = amountWethToSwap * minETHLUSDRate; //WETH and LUSD is 18 decimals
+
+                // From https://docs.uniswap.org/protocol/guides/swaps/multihop-swaps#calling-the-function-1
+                // Multiple pool swaps are encoded through bytes called a `path`. A path is a sequence of token addresses and poolFees that define the pools used in the swaps.
+                // The format for pool encoding is (tokenIn, fee, tokenOut/tokenIn, fee, tokenOut) where tokenIn/tokenOut parameter is the shared token across the pools.
+                // Since we are swapping WETH to DAI and then DAI to LUSD the path encoding is (WETH, 0.3%, DAI, 0.3%, LUSD).
+                ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams({
+                    path: abi.encodePacked(wethAddress, poolFee, hopTokenAddress, poolFee, address(_tokens[0])),
+                    recipient: address(this),
+                    deadline: block.timestamp,
+                    amountIn: amountWethToSwap,
+                    amountOutMinimum: amountLUSDMin
+                });
+
+                // Executes the swap
+                gain = uint128(swapRouter.exactInput(params));
+                if (gain > 0) {
+                    swappedLUSDSuccessfully = true;
+                }
+            }
+        }
+
+        // If swap was successful (or if percent to swap is 0), send the remaining WETH to the treasury.  Crucial check otherwise we'd send all our WETH to the treasury and not respect our desired percentage
+        if (ethToLUSDRatio == 0 || swappedLUSDSuccessfully) {
+            uint256 wethBalance = IWETH(wethAddress).balanceOf(address(this));
+            if (wethBalance > 0) {
+                IERC20(wethAddress).safeTransfer(address(treasury), wethBalance);
+            }
+        }
+
+        // 4.  Deposit all LUSD in balance to into StabilityPool.
+        uint256 lusdBalance = _tokens[0].balanceOf(address(this));
+        if (lusdBalance > 0) {
+            lusdStabilityPool.provideToSP(lusdBalance, frontEndAddress); //s either a front-end address OR 0x0
+        }
+    }
+
+    function deallocate(uint256[] memory amounts) public override {
```

Comment:
yea i wanted to do this but i though the amounts[] array was meant to be tied to _tokens in base allocator for some reason. Because LQTY is not part of _tokens right? since its a reward token not a allocator token.

From what youre saying we can make amounts[] [lusd, LQTY]. despite LQTY not being in _tokens?

Reply:
I think we can be relatively lax with it. You are right in that the idea is that utility or reward tokens should be either idle or harvested with the contract so there should be no explicit need to unstake the tokens, when I actually think about it since LQTY here is being used for farming in reality LQTY should have also been under _tokens, and then another id registered, but not to overcomplicate I think it can be left as is and what I suggested implemented based on need, e.g. if you think that a certain amount of LQTY might just be too much.